app/controllers/notes_controller.rb

```diff
@@ -106,11 +113,12 @@ def update
 
   def destroy
     @note = Note.find(params[:id])
-    if @note.user_can_modify?(current_user)
+    begin
+      authorize! @note, to: :ensure_can_modify?
       @note.destroy
-      respond_with @note
-    else
-      flash_message(:error, I18n.t('notes.delete.error_permissions'))
+      respond_with(@note)
+    rescue ActionPolicy::Unauthorized
+      flash_now(:error, t('action_policy.policy.note.ensure_can_modify?'))
```

Comment:
Can we use the actual error message here. This will make things easier if we decide to change the error message in the future.